New component: `discovery.nomad` (#4550)
grafana/agent: docs/sources/flow/reference/components/discovery.nomad.md

---
canonical: https://grafana.com/docs/agent/latest/flow/reference/components/discovery.nomad/
title: discovery.nomad
---

# discovery.consul

`discovery.nomad` allows you to retrieve scrape targets from [Nomad's](https://www.nomadproject.io/) Service API.

## Usage

```river
discovery.nomad "LABEL" {
  server = NOMAD_SERVER
}
```

## Arguments

The following arguments are supported:

Name | Type | Description | Default | Required
---- | ---- | ----------- | ------- | --------
`server` | `string` | Address of nomad server. | `http://localhost:4646` | no
`namespace` | `string` | Nomad namespace to use. | `default` | no
`region` | `string` | Nomad region to use. | `global` | no
`allow_stale` | `bool` | Allow reading from non-leader nomad instances. | `true` | no
`tag_separator` | `string` | Seperator to join nomad tags into Prometheus labels. | `,` | no

> 

Reviewer: this newline breaks the table

oauth2 | [oauth2][] | Configure OAuth2 for authenticating to the endpoint. | no
oauth2 > tls_config | [tls_config][] | Configure TLS settings for connecting to the endpoint. | no

The `>` symbol indicates deeper levels of nesting. For example,
`oauth2 > tls_config` refers to a `tls_config` block defined inside
an `oauth2` block.

[basic_auth]: #basic_auth-block
[authorization]: #authorization-block
[oauth2]: #oauth2-block
[tls_config]: #tls_config-block

### basic_auth block

{{< docs/shared lookup="flow/reference/components/basic-auth-block.md" source="agent" >}}

### authorization block

{{< docs/shared lookup="flow/reference/components/authorization-block.md" source="agent" >}}

### oauth2 block

{{< docs/shared lookup="flow/reference/components/oauth2-block.md" source="agent" >}}

### tls_config block

{{< docs/shared lookup="flow/reference/components/tls-config-block.md" source="agent" >}}

## Exported fields

The following fields are exported and can be referenced by other components:

Name | Type | Description
---- | ---- | -----------
`targets` | `list(map(string))` | The set of targets discovered from the nomad server.

Each target includes the following labels:

> TODO:

Reviewer: is this TODO intended?

## Usage

```river
discovery.nomad "LABEL" {
  server = NOMAD_SERVER
}
```

## Arguments

The following arguments are supported:

Name | Type | Description | Default | Required
---- | ---- | ----------- | ------- | --------
`server` | `string` | Address of nomad server. | `http://localhost:4646` | no
`namespace` | `string` | Nomad namespace to use. | `default` | no
`region` | `string` | Nomad region to use. | `global` | no
`allow_stale` | `bool` | Allow reading from non-leader nomad instances. | `true` | no
`tag_separator` | `string` | Seperator to join nomad tags into prometheus labels. | `,` | no

`refresh_interval` | `duration` | Frequency to refresh list of containers. | `"30s"` | no
`bearer_token` | `secret` | Bearer token to authenticate with. | | no
`bearer_token_file` | `string` | File containing a bearer token to authenticate with. | | no
`proxy_url` | `string` | HTTP proxy to proxy requests through. | | no
`follow_redirects` | `bool` | Whether redirects returned by the server should be followed. | `true` | no
`enable_http2` | `bool` | Whether HTTP2 is supported for requests. | `true` | no

>  At most one of the following can be provided:

Reviewer 2: Are these optional? Can you skip this? When it says `can be provided`, it implies it's optional. So checking to make sure this is intended to be optional.

```suggestion
 You can provide one of the following arguments for authentication:
```

Or if it's mandatory:

> You must supply one of the following arguments for authentication:
